index.js

```diff
@@ -1,88 +1,239 @@
-        {
-          pattern: '~*/**',
-          group: 'external',
-          position: 'before'
-        },
-        {
-          pattern: '@/composables/**',
-          group: 'parent',
-          position: 'before'
-        },
-        {
-          pattern: '@/components/**',
-          group: 'parent',
-          position: 'before'
-        },
-        {
-          pattern: '@/**',
-          group: 'internal',
-          position: 'after'
+/**
+ * @typedef {Parameters<typeof antfu>[1]} UserConfig
+ * @param {...UserConfig} userConfigs - User-defined ESLint configurations to extend or override
+ * @returns {ReturnType<typeof antfu>} A chainable ESLint config
+ */
+export default function globalbrain(...userConfigs) {
+  return (
+    antfu(
+      {
+        // `type: 'lib'` only adds [one rule](https://github.com/antfu/eslint-config/blob/0cd12cc90d2100798a5d8f5d51b34753b7be7f70/src/configs/typescript.ts#L165-L173),
+        // as our projects are mostly apps, we can just use the default app config,
+        // and either provide a separate config for lib projects,
+        // or manually enable the rule in the projects if needed
+
+        // TS/Vue features are auto-detected,
+        // but to avoid errors being thrown from our overrides,
+        // we need to explicitly enable them here.
+        typescript: true,
+        vue: true,
+
+        // We are more conservative on curly braces
+        // <https://github.com/antfu/eslint-config/blob/0cd12cc90d2100798a5d8f5d51b34753b7be7f70/src/configs/stylistic.ts#L54-L63>
+        lessOpinionated: true
+      },
+      {
+        rules: {
+          'max-statements-per-line': 'off'
         }
-      ],
-      alphabetize: {
-        order: 'asc',
-        caseInsensitive: false
       }
-    }],
-    'new-cap': ['error', {
-      newIsCap: false,
-      capIsNew: false,
-      properties: true
-    }],
-    'n/prefer-global/process': 'off',
+    )
+      .override('antfu/javascript/rules', {
+        rules: {
+          'no-cond-assign': 'off',
+          'new-cap': [
+            'error',
+            {
+              newIsCap: false,
+              capIsNew: false,
+              properties: true
+            }
+          ]
+        }
+      })
+      .override('antfu/stylistic/rules', {
+        rules: {
+          'style/arrow-parens': ['error', 'always'],
+          'style/brace-style': [
+            'error',
+            '1tbs',
+            {
+              allowSingleLine: true
+            }
+          ],
+          'style/comma-dangle': ['error', 'never']
+        }
+      })
+      .override('antfu/typescript/rules', {
+        rules: {
+          'ts/consistent-type-definitions': 'off',
+          'ts/consistent-type-imports': [
+            'error',
+            {
+              prefer: 'type-imports',
+              disallowTypeAnnotations: false,
+              fixStyle: 'inline-type-imports'
+            }
+          ],
+          'ts/no-empty-interface': 'off',
```

Comment:
this is disabled in latest version? replaced by no-empty-object-type which I think is fine to enable.